glib/src/source.rs

```diff
@@ -827,44 +827,54 @@ where
 
 // rustdoc-stripper-ignore-next
 /// The priority of sources
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub struct Priority(i32);
+#[derive(Default, Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Priority {
+    #[doc(alias = "G_PRIORITY_HIGH")]
+    High,
+    #[doc(alias = "G_PRIORITY_DEFAULT")]
+    #[default]
+    Default,
+    #[doc(alias = "G_PRIORITY_HIGH_IDLE")]
+    HighIdle,
+    #[doc(alias = "G_PRIORITY_DEFAULT_IDLE")]
+    DefaultIdle,
+    #[doc(alias = "G_PRIORITY_LOW")]
+    Low,
+    Custom(i32),
```

Comment:
Not sure if that matters here but this increases the size of the type from 32 bits to 40 bits, which means that it e.g. will require two registers on 32 bit platforms and usually will require 64 bits of storage (padding, etc).

You might instead want to keep the previous definition and add associated constants instead? Then you can still have `Priority::DEFAULT`, `Priority::HIGH`, etc.

Reply:
Interesting. Switched to using associated consts instead

Reply:
Ideally Rust would allow defining enums in such a way that there are a few fixed values and all others are mapped to a generic one, but that's not possible right now except for things inside std.